fix: ServerAuthContext types not augmented in defineServerAuth

## Problem

`ctx.runtimeConfig` and `ctx.db` in `defineServerAuth` have fallback types (`Record<string, unknown>` and `unknown`) instead of the augmented types (`RuntimeConfig` and Drizzle db type).

**Credits:** Issue discovered by @amaury-tobias in #26. Their approach was incorrect but the underlying DX problem is real.

## Reproduction

```bash
git clone --depth 1 --filter=blob:none --sparse https://github.com/onmax/repros.git
cd repros && git sparse-checkout set nuxt-better-auth-26
cd nuxt-better-auth-26 && pnpm i
BETTER_AUTH_SECRET="12345678901234567890123456789012" pnpm prepare && pnpm typecheck
```

**Error:**
```
server/auth.config.ts(7,9): error TS2739: Type 'Record<string, unknown>' is missing the following properties from type 'RuntimeConfig': app, public, hub, betterAuthSecret, auth
```

## Expected

- `ctx.runtimeConfig` → `RuntimeConfig` from `nuxt/schema`
- `ctx.db` → Drizzle D1 database type

## Actual

- `ctx.runtimeConfig` → `Record<string, unknown>`
- `ctx.db` → `unknown`

## Root Cause

`defineServerAuth` in `config.ts` imports `ServerAuthContext` directly from `./types/augment.js`:

```typescript
import type { ServerAuthContext } from './types/augment'
export function defineServerAuth<T>(config: (ctx: ServerAuthContext) => T): ...
```

The generated types in `.nuxt/types/nuxt-better-auth-infer.d.ts` use `declare module '#nuxt-better-auth'` to extend the interface, but `config.ts` imports from the source file directly - not from the module alias.

**TypeScript declaration merging doesn't work across different import paths.**

## Why PR #26 is Wrong

PR #26 tries to fix this by changing `augment.ts` directly with:
```typescript
import type { db } from '#imports'
import type { NitroRuntimeConfig } from 'nitropack/types'
```

This is incorrect because:
1. `#imports` is a Nuxt runtime alias - doesn't exist in module source code at build time
2. Would break for users without NuxtHub/db configured
3. Violates the fallback/augmentation pattern

--- a/src/module.ts
+++ b/src/module.ts
@@ -190,6 +190,18 @@ declare module '#nuxt-better-auth' {
     ${hasHubDb ? `db: typeof import('hub:db')['db']` : ''}
   }
 }
+
+// Augment the config module to use the extended ServerAuthContext
+interface _AugmentedServerAuthContext {
+  runtimeConfig: RuntimeConfig
+  ${hasHubDb ? `db: typeof import('hub:db')['db']` : 'db: unknown'}
+}
+
+declare module '@onmax/nuxt-better-auth/config' {
+  import type { BetterAuthOptions } from 'better-auth'
+  type ServerAuthConfig = Omit<BetterAuthOptions, 'database' | 'secret' | 'baseURL'>
+  export function defineServerAuth<T extends ServerAuthConfig>(config: (ctx: _AugmentedServerAuthContext) => T): (ctx: _AugmentedServerAuthContext) => T
+}
 `,
     })
 

